pipeline_templates/codeql3000_default.yml

```diff
@@ -0,0 +1,80 @@
+# This is the job template for running CodeQL on a repository
+
+parameters:
+- name: 'cmake_command'
+  type: string
+  default: '.. -G "Visual Studio 17 2022" -A x64'
+
+jobs:
+- job: codeql
+  displayName: 'Run CodeQL on production code'
+  pool:
+    name: Azure-MessagingStore-WinBuildPoolVS2022
+    demands:
+    - Cmd
+    - msbuild
+    - visualstudio
+    - build
+
+  workspace:
+    clean: all
+
+  steps:
+  - task: BatchScript@1
+    displayName: 'Git submodule update'
+    inputs:
+      filename: 'C:\Program Files\Git\bin\git.exe'
+      arguments: 'submodule update --init --force'
+
+  - task: BatchScript@1
+    displayName: 'Git submodule clean'
+    inputs:
+      filename: 'C:\Program Files\Git\bin\git.exe'
+      arguments: 'submodule foreach --recursive "git clean -xdff"'
+
+  - task: BatchScript@1
+    displayName: 'Git clean'
+    inputs:
+      filename: 'C:\Program Files\Git\bin\git.exe'
+      arguments: 'clean -xdff'
+
+  - task: DeleteFiles@1
+    displayName: 'Clean build binaries directory'
+    inputs:
+      sourceFolder: $(Build.BinariesDirectory)
+      contents: \*
+
+  - task: BatchScript@1
+    displayName: 'Setup VS Vars'
+    inputs:
+      filename: '"c:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\VsDevCmd.bat"'
+      modifyEnvironment: true
+
+  - template: codeql3000_init.yml
+
+  - task: CMake@1
+    displayName: 'CMake ${{parameters.cmake_command}}'
+    inputs:
+      workingDirectory: 'build_x64'
+      cmakeArgs: ${{parameters.cmake_command}}
+
+  - task: VSBuild@1
+    displayName: 'Build solution build_x64\*.sln'
+    inputs:
+      solution: 'build_x64\*.sln'
+      msbuildArgs: '/t:restore /t:build'
+      platform: x64
+      configuration: RelWithDebInfo
```

Comment:
>RelWithDebInfo [](http://example.com/codeflow?start=21&length=14)

Looks like we should prevent this template from being invoked from a "Debug" build configuration.

Maybe add build_configuration as parameter and immediately exit the template if it is not RelWithDebInfo?

And what's wrong with running codeql for Debug? Asking for a friend...

Reply:
Made this an parameter. There's nothing wrong with Debug, just took whatever we were running already in some of the repos.